cmd/history.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"time"
+	"strings"
+
+	"github.com/dustin/go-humanize"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/backend"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/diag/colors"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+)
+
+func newHistoryCmd() *cobra.Command {
+	var stack string
+	var cmd = &cobra.Command{
+		Use:        "history",
+		Aliases:    []string{"hist"},
+		SuggestFor: []string{"updates"},
+		Short:      "Update history for a stack",
+		Long: "Update history for a stack\n" +
+		"\n" +
+		"This command lists data about previous updates for a stack.",
+		Args: cmdutil.NoArgs,
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+
+			opts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+
+			s, err := requireStack(stack, false, opts, false /*setCurrent*/)
+
+			if err != nil {
+				return err
+			}
+
+			b := s.Backend()
+
+			updates, err := b.GetHistory(commandContext(), s.Ref())
+			if err != nil {
+				return errors.Wrap(err, "getting history")
+			}
+
+			displayUpdate(updates, opts)
+
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"Choose an stack other than the currently selected one")
+
+	return cmd
+}
+
+func displayUpdate(updates []backend.UpdateInfo, opts display.Options) {
+
+	if len(updates) == 0 {
+		fmt.Println("Stack has never been updated")
+		return
+	}
+
+	printResourceChanges := func(backgroundColor string, textColor string, sign string, amountOfChange int, resetColor string) {
+		msg := opts.Color.Colorize(fmt.Sprintf("%s%s%s%v%s", backgroundColor, textColor, sign, amountOfChange, resetColor))
+		fmt.Print(msg)
+	}
+
+	for _, update := range updates {
+
+		fmt.Print( opts.Color.Colorize( fmt.Sprintf("%s%s%s\n", colors.Green, "-----------------------", colors.Reset)))
```

Comment:
1. unnecessary space after `Print(` and `Colorize(`.  I'm actually suprised the linter didn't complain.